fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -1164,7 +1164,7 @@ private ChangedTransactionDetail reprocessChangedLoanTransactions(Loan loan,
 
     @Transactional
     @Override
-    public Loan makeAccrualActivityTransaction(Loan loan, final LoanRepaymentScheduleInstallment installment,
+    public void makeAccrualActivityTransaction(Loan loan, final LoanRepaymentScheduleInstallment installment,
```

Comment:
If we are explicitly saving the loan again, returning the Loan entity would be advisable hence during merge the original instance got detached.